src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/RSSAnalyzer.scala

```diff
@@ -13,21 +13,40 @@ class RSSAnalyzer extends Analyzer[RSSEntry] with Serializable with AnalysisDefa
 
   override def toSchema(item: RSSEntry, locationFetcher: LocationFetcher, imageAnalyzer: ImageAnalyzer): ExtendedDetails[RSSEntry] = {
     val body = readDescription(item)
+    val url = getSourceUrlFromItem(item)
     ExtendedDetails(
-      eventid = s"RSS.${item.uri}",
+      eventid = s"RSS.${url.getOrElse(item.uri)}",
       sourceeventid = item.uri,
       eventtime = item.publishedDate,
       body = body,
       imageurl = None,
       title = item.title,
       externalsourceid = item.source.uri,
       pipelinekey = "RSS",
-      sourceurl = item.uri,
+      sourceurl = url.getOrElse(""),
```

Comment:
Hopefully pipeline doesn't validate this one is non empty? Empty string won't pass it if it's checking this field too.

Reply:
I don't think it does, but the UI does use this URL. In any case, the absence of this value is pathological/unlikely.